go/gilded-rose.go

```diff
@@ -8,51 +12,47 @@ type Item struct {
 func UpdateQuality(items []*Item) {
 	for i := 0; i < len(items); i++ {
 
-		if items[i].name != "Aged Brie" && items[i].name != "Backstage passes to a TAFKAL80ETC concert" {
-			if items[i].quality > 0 {
-				if items[i].name != "Sulfuras, Hand of Ragnaros" {
-					items[i].quality = items[i].quality - 1
-				}
+		if items[i].name == "Aged Brie" {
+			IncreaseQualityByOne(items[i])
+			if items[i].sellIn <= 0 {
+				IncreaseQualityByOne(items[i])
 			}
-		} else {
-			if items[i].quality < 50 {
-				items[i].quality = items[i].quality + 1
-				if items[i].name == "Backstage passes to a TAFKAL80ETC concert" {
-					if items[i].sellIn < 11 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
-					if items[i].sellIn < 6 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
+		} else if items[i].name == "Backstage passes to a TAFKAL80ETC concert" {
+			if items[i].sellIn > 0 {
+				IncreaseQualityByOne(items[i])
+				if items[i].sellIn < 6 {
+					IncreaseQualityByOne(items[i])
 				}
+				if items[i].sellIn < 11 {
+					IncreaseQualityByOne(items[i])
+				}
+
+			} else {
+				items[i].quality = 0
+			}
+
+		} else if strings.Contains(items[i].name, "Conjured") {
+			if items[i].quality > 0 {
+				items[i].quality = items[i].quality - 2
+			}
+		} else if items[i].name != "Sulfuras, Hand of Ragnaros" {
```

Comment:
Code coverage shows that this block is not covered by the test cases.

Reply:
yes, I did in the last moment and forgot to push it